Sources/BBCodeParser.php

```diff
@@ -2940,7 +2940,7 @@ function ($matches) {
 							return $url;
 						}
 
-						return '[url=&quot;' . str_replace(['[', ']'], ['&#91;', '&#93;'], $full_url->toAscii()) . '&quot;]' . $url . '[/url]';
+						return '[url=&quot;' . str_replace(['[', ']'], ['&#91;', '&#93;'], strval($full_url->toAscii())) . '&quot;]' . $url . '[/url]';
```

Comment:
This isn't necessary.

Reply:
This was mostly a complaint by static analysis.  Because toAscii() returns a object: `	public function toAscii(): object`.  So static analysis says you can't use a object in str_replace for the subject.  `strval` can take a object in and is going to return a string.  We could expose a the __toString as a non magic method.  But as well, toAscii seems like it should return a string, but I couldn't be sure why we are still returning the object.

Reply:
The static analysis is mistaken in this case. An instance of Stringable will be cast to string automatically when used as a string by str_replace().

The toAscii() method changes the Url object itself. It returns $this for the sake of method chaining.

Reply:
Well I don't think relying on the __ToString magic method to stringify the object is the best practice to do for production code.  But I reverted the change.